st2common/st2common/services/action.py

```diff
@@ -260,6 +260,12 @@ def request_cancellation(liveaction, requester):
 
     # Run cancelation sequence for liveaction that is in running state or
     # if the liveaction is operating under a workflow.
+    if 'parent' not in liveaction.context and action_constants.LIVEACTION_STATUS_RUNNING:
```

Comment:
I can't tell why parent not in liveaction.context is related to the issue we're fixing here.  Can you explain?